Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -1404,6 +1406,28 @@ private static void MakeSharedOpaqueOutputIfNeeded(IPipExecutionEnvironment envi
 
                                 ++retryCount;
 
+                                if (result.PipProperties != null)

```

Comment:
>if [](start = 32, length = 2)

This is my suggestion to make the code less confusing (at least for me):
```
// Declare aggregate pip properties before the while loop.
var aggregatePipProperties = new Dictionary<string, int>(); // If memory is a concern, we can use object pool -- we have object pool.

while (true) 
{
...
    if (result.PipProperties != null) {
        foreach (var kvp in result.PipProperties) {
             if (aggregatePipProperties.TryGetValue(kvp.Key, out var value)) {
                 aggregatePipProperties[kvp.Key] = value + kvp.Value;
             } else {
                 aggregatePipProperties.Add(kvp.Key, kvp.Value);
             }
        }
    }
}

if (aggregatePipProperties.Count > 0) {
    // The aggregate result should override the last result.
    result.PipProperties = aggregatePipProperties;
}

```

What do you think?

Reply:
Sounds good to me, updated.